### PR TITLE
Docs for token extraction for Apple users

### PR DESCRIPTION
**Pull request by domibarton**
Added description how to extract SQLite database from iOS backup & manually extract/decrypt tokens from SQLite database.

### CODE REVIEW

File: docs/discovery.rst [hunk 1]
```diff
@@ -217,6 +237,29 @@ to the tool with ``--password <password>``.
             Token: 476e6b70343055483XXX
             MAC: 28:6C:07:XX:XX:XX
 
+Extracting tokens manually
```
[the remark this answers is not in the document]
Comment (domibarton) [reply docs/discovery.rst]: It didn't work for my sqlite database and therefor I've added this part. There was a `TypeError` exception:

```
(homeassistant) root@rpi:/opt/homeassistant# miio-extract-tokens /tmp/1672094281_mihome.sqlite
INFO:miio.extract_tokens:Reading database from /tmp/1672094281_mihome.sqlite
INFO:miio.extract_tokens:Reading tokens from Apple DB
Traceback (most recent call last):
  File "/opt/homeassistant/homeassistant/bin/miio-extract-tokens", line 11, in <module>
    sys.exit(main())
  File "/opt/homeassistant/homeassistant/lib/python3.5/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/opt/homeassistant/homeassistant/lib/python3.5/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/opt/homeassistant/homeassistant/lib/python3.5/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homeassistant/homeassistant/lib/python3.5/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/opt/homeassistant/homeassistant/lib/python3.5/site-packages/miio/extract_tokens.py", line 155, in main
    devices = list(reader.read_tokens(backup))
  File "/opt/homeassistant/homeassistant/lib/python3.5/site-packages/miio/extract_tokens.py", line 113, in read_tokens
    yield from self.read_apple()
  File "/opt/homeassistant/homeassistant/lib/python3.5/site-packages/miio/extract_tokens.py", line 74, in read_apple
    token = BackupDatabaseReader.decrypt_ztoken(dev['ZTOKEN'])
  File "/opt/homeassistant/homeassistant/lib/python3.5/site-packages/miio/extract_tokens.py", line 50, in decrypt_ztoken
    if len(ztoken) <= 32:
TypeError: object of type 'NoneType' has no len()
```

[the remark this answers is not in the document]
Comment (rytilahti) [reply docs/discovery.rst]: This should obviously be fixed. Could you try with `--dump-raw` to see what it outputs?

[the remark this answers is not in the document]
Comment (domibarton) [reply docs/discovery.rst]: > This should obviously be fixed. Could you try with `--dump-raw` to see what it outputs?

Sorry busy again :) I can test it tonight… let me open an issue for that!

[the remark this answers is not in the document]
Comment (rytilahti) [reply docs/discovery.rst]: Thanks! Let's merge this now, it's a good idea still have the manual way documented, too :-)

**Pull request merged**